packages/compiler/src/core/checker.ts

```diff
@@ -5323,8 +5323,16 @@ export function createChecker(program: Program): Checker {
           );
         }
       } else {
+        // It is possible that the property type is a reference to another
```

Comment:
That fix this issue but wondering if we should not be just fixing it globally

```
model Foo {
  bar: string;
}

// Should Foo.bar be assignable to string all together

model Temp<T extends string> {

}

Temp<Foo.Ba> // should that be valid
```

I think so but either way this also opens up another range of case to handle in the decorators. You have to make sure you handle a model property all the type for whatever you are pased

Reply:
Hmmm, is there a case where we'd actually want the `ModelProperty` to come through as a decorator parameter or should we always just resolve its type and pass it along?  I can imagine some cases where handling the `ModelProperty` directly could be useful, but I dunno if it's too easy to abuse?

Reply:
yeah this is a good point, we do have some precendent depending on what you are asking for it might or not cast the value

```
dec foo(value: valueof string) // js string
dec foo(value: string) // StringLiteral
```

but maybe we need a design meeting here to approve this, is this change urgent as it opens this behavior and changing it would be a breaking change

Reply:
It's not urgent but it will be necessary to enable a partner team to follow some guidance I'm preparing for them for library design.  Not a P0 blocker at the moment, but something we'll need to figure out soon.  I'm fine with having a design meeting about it to discuss a good general-case option

Reply:
Yeah talking to @bterlson about this too doing the auto casting might not be a good solution either so be good to have a design meeting to talk about the options here.

Issue with the auto casting is you might be ignoring information on the property that could be relevant. A user might expect that to work and it could result in confusing experience.